### PR TITLE
MQTT: add support for custom ca cert

### PR DESCRIPTION
**Pull request by pailloM**
add extra option to set path of CA cert for self-signed certificate

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
Thank you very much for your contribution!
Would you mind creating the PR against the Dev branch?

**Comment by pailloM** [comment]
Changed to the dev branch. 
Let me know if you want any other changes

File: src/HABApp/mqtt/mqtt_connection.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import logging
+import ssl
```
Comment (spacemanspiff2007) [inline src/HABApp/mqtt/mqtt_connection.py:2]: Could you remove the unnecessary import please?

**Comment by spacemanspiff2007** [comment]
Thanks - almost there:
```
 ./src/HABApp/mqtt/mqtt_connection.py:2:1: F401 'ssl' imported but unused
```

**Comment by pailloM** [comment]
done

**Comment by spacemanspiff2007** [comment]
Thank you very much!

**Comment by spacemanspiff2007** [comment]
@pailloM :
I renamed the parameter to ``tls_ca_cert_path`` and added a description to the config file.
What do you think? Do you pass a folder or a file?

**Comment by pailloM** [comment]
I pass a file with it's full path. For example:
/opt/HABApp/conf/ca.crt

**Comment by spacemanspiff2007** [comment]
I can't seem to make up my mind about the parameter. Now it's ``tls_ca_cert``.
I've added a test to see if the file exists.
What do you think?
Would you mind testing the dev branch?

**Comment by pailloM** [comment]
Sure. Will give it a spin right now

**Comment by pailloM** [comment]
I get the following error: 
ValueError: SSL/TLS has already been configured

line 59 is not required anymore and should be removed. Without it, all working fine.
`mqtt_client.tls_set() ` will be called in both cases, with and without the certificate being specified.

You might want to raise an error if the file does not exist (line 65). Something like:
raise FileNotFoundError(ca_cert)

**Comment by spacemanspiff2007** [comment]
@pailloM 
I fixed it and prevent connecting on missing certificate.
It would be nice if you could take a look at it again.

**Comment by pailloM** [comment]
I still get ValueError: SSL/TLS has already been configured
mqtt_client.tls_set() is called twice when the certificate is set. Once on line 69 and a second time on line 72.

line 71-72 should be:
`
# enable TLS after (!) we set the certificate
else:
    mqtt_client.tls_set()       
`

The comment should read something like:
# enable TLS when the certificate is not set

**Comment by spacemanspiff2007** [comment]
Yes, how could I missed this after you made it so clear. Sorry!
I've pushed a fix (this time for real!). Sorry for your efforts.

**Comment by pailloM** [comment]
Checked. All working!
Thanks for adding that

**Comment by spacemanspiff2007** [comment]
Thank you for your contribution and your testing!
I'll make a new release this week which will include your changes!

**Pull request merged**